Description: Add support to kFreeBSD
Author: Steven Chamberlain <steven@pyro.eu.org>
Last-Update: 2016-12-04
Index: hashcat-3.20/include/common.h
===================================================================
--- hashcat-3.20.orig/include/common.h
+++ hashcat-3.20/include/common.h
@@ -12,7 +12,7 @@
 #define _POSIX
 #elif defined (__APPLE__)
 #define _POSIX
-#elif defined (__FreeBSD__)
+#elif defined (__FreeBSD_kernel__)
 #define _POSIX
 #elif defined (_WIN32) || defined (_WIN64)
 #define _WIN 1
Index: hashcat-3.20/include/ext_OpenCL.h
===================================================================
--- hashcat-3.20.orig/include/ext_OpenCL.h
+++ hashcat-3.20/include/ext_OpenCL.h
@@ -21,7 +21,7 @@
 #include <CL/cl.h>
 #endif
 
-#if defined (__FreeBSD__)
+#if defined (__FreeBSD_kernel__)
 #include <CL/cl.h>
 #endif
 
Index: hashcat-3.20/src/Makefile
===================================================================
--- hashcat-3.20.orig/src/Makefile
+++ hashcat-3.20/src/Makefile
@@ -14,6 +14,9 @@ UNAME                   := $(shell uname
 # we need to strip the windows version number to be able to build hashcat on cygwin hosts
 UNAME                   := $(patsubst CYGWIN_NT-%,CYGWIN,$(UNAME))
 
+# :XXX: treat GNU/kFreeBSD, Hurd and Linux
+UNAME                    := $(patsubst GNU%,Linux,$(UNAME))
+
 # same for msys
 UNAME                   := $(patsubst MSYS_NT-%,MSYS2,$(UNAME))
 UNAME                   := $(patsubst MINGW32_NT-%,MSYS2,$(UNAME))
@@ -72,8 +75,8 @@ SED                     := gsed
 endif
 
 ifeq ($(UNAME),FreeBSD)
-CC                      := cc
-SED                     := gsed
+CC                      := gcc
+SED                     := sed
 endif
 
 ##
@@ -165,7 +168,6 @@ endif # Linux
 ifeq ($(UNAME),FreeBSD)
 CFLAGS_NATIVE           := $(CFLAGS)
 CFLAGS_NATIVE           += -I$(OPENCL_HEADERS_KHRONOS)/
-CFLAGS_NATIVE           += -march=native
 LFLAGS_NATIVE           := $(LFLAGS)
 LFLAGS_NATIVE           += -lpthread
 endif # FreeBSD
Index: hashcat-3.20/src/shared.c
===================================================================
--- hashcat-3.20.orig/src/shared.c
+++ hashcat-3.20/src/shared.c
@@ -7,6 +7,11 @@
 #include "types.h"
 #include "shared.h"
 
+#if defined(__FreeBSD__) || defined(__FreeBSD_kernel__)
+#include <sys/types.h>
+#include <sys/sysctl.h>
+#endif
+
 bool is_power_of_2 (const u32 v)
 {
   return (v && !(v & (v - 1)));
@@ -250,4 +255,4 @@ void setup_seeding (const bool rp_gen_se
 
     srand (ts);
   }
-}
\ No newline at end of file
+}
Index: hashcat-3.20/include/sort_r.h
===================================================================
--- hashcat-3.20.orig/include/sort_r.h
+++ hashcat-3.20/include/sort_r.h
@@ -25,7 +25,7 @@ Slightly modified to work with hashcat t
 */
 
 #if (defined __APPLE__ || defined __MACH__ || defined __DARWIN__ || \
-     defined __FreeBSD__ || defined __DragonFly__)
+     defined __FreeBSD__ || defined __FreeBSD_kernel__ || defined __DragonFly__)
 #  define _SORT_R_BSD
 #  define _SORT_R_INLINE inline
 #elif (defined __linux__) || defined (__CYGWIN__)
Index: hashcat-3.20/src/folder.c
===================================================================
--- hashcat-3.20.orig/src/folder.c
+++ hashcat-3.20/src/folder.c
@@ -48,7 +48,7 @@ static int get_exec_path (char *exec_pat
 
   const size_t len = strlen (exec_path);
 
-  #elif defined (__FreeBSD__)
+  #elif defined (__FreeBSD__) || defined(__FreeBSD_kernel__)
 
   #include <sys/sysctl.h>
 
@@ -65,9 +65,7 @@ static int get_exec_path (char *exec_pat
 
   sysctl (mib, 4, exec_path, &size, NULL, 0);
 
-  const ssize_t len = readlink (tmp, exec_path, exec_path_sz - 1);
-
-  if (len == -1) return -1;
+  const size_t len = strlen (exec_path);
 
   #else
   #error Your Operating System is not supported or detected
