From: Sophie Brun <sophie@freexian.com>
Date: Thu, 18 Jun 2020 10:42:18 +0200
Subject: Fix spelling error

Last-Update: 2020-06-18
---
 src/user_options.c | 1 -
 1 file changed, 1 deletion(-)

diff --git a/src/user_options.c b/src/user_options.c
index 4dd2dce..f247db7 100644
--- a/src/user_options.c
+++ b/src/user_options.c
@@ -119,7 +119,6 @@ static const struct option long_options[] =
   {"segment-size",              required_argument, NULL, IDX_SEGMENT_SIZE},
   {"self-test-disable",         no_argument,       NULL, IDX_SELF_TEST_DISABLE},
   {"separator",                 required_argument, NULL, IDX_SEPARATOR},
-  {"seperator",                 required_argument, NULL, IDX_SEPARATOR},
   {"session",                   required_argument, NULL, IDX_SESSION},
   {"show",                      no_argument,       NULL, IDX_SHOW},
   {"skip",                      required_argument, NULL, IDX_SKIP},
