Description: update the makefile Change CFLAGS
Author: Sophie Brun <sophie@freexian.com>
Origin: vendor
Last-Update: 2016-06-30
Index: hashcat/src/Makefile
===================================================================
--- hashcat.orig/src/Makefile
+++ hashcat/src/Makefile
@@ -173,12 +173,12 @@ endif
 ##
 
 ifeq ($(UNAME),Linux)
-CFLAGS_NATIVE           := $(CFLAGS)
+CFLAGS_NATIVE           += $(CFLAGS) $(CPPFLAGS)
 CFLAGS_NATIVE           += -I$(OPENCL_HEADERS_KHRONOS)/
 CFLAGS_NATIVE           += -march=native
 CFLAGS_NATIVE           += -DWITH_HWMON
 LFLAGS_NATIVE           := $(LFLAGS)
-LFLAGS_NATIVE           += -lpthread -ldl
+LFLAGS_NATIVE           += -lpthread -ldl $(LDFLAGS)
 endif # Linux
 
 ifeq ($(UNAME),FreeBSD)
