Description: update the makefile Change CFLAGS
Author: Sophie Brun <sophie@freexian.com>
Origin: vendor
Last-Update: 2016-06-30
--- a/src/Makefile
+++ b/src/Makefile
@@ -87,7 +87,7 @@ VERSION_TAG              := $(shell test
 ## Compiler flags
 ##
 
-CFLAGS                   := -pipe -W -Wall -std=c99 -Iinclude/ -IOpenCL/ -I$(OPENCL_HEADERS_KHRONOS)/
+CFLAGS                   += -pipe -W -Wall -std=c99 -Iinclude/ -IOpenCL/ -I$(OPENCL_HEADERS_KHRONOS)/
 
 ifndef DEBUG
 CFLAGS                   += -O2
@@ -117,8 +117,8 @@ CFLAGS_NATIVE            := -D_POSIX -DL
 ifndef DEBUG
 CFLAGS_NATIVE            += -s
 endif
-CFLAGS_NATIVE            += $(CFLAGS)
-LFLAGS_NATIVE            := -lpthread -ldl
+CFLAGS_NATIVE            += $(CFLAGS) $(CPPFLAGS)
+LFLAGS_NATIVE            := -lpthread -ldl $(LDFLAGS)
 CFLAGS_NATIVE            += -DHAVE_HWMON
 endif # linux
 
